store/proof.go

```diff
@@ -0,0 +1,178 @@
+package store
+
+import (
+	"fmt"
+
+	"github.com/cometbft/cometbft/crypto/merkle"
+	cmtcrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
```

Comment:
we can get rid of the cometbft import here, there isnt a need to wrap it. we dont register this on comet like we did preprotobuf so this is legacy, anyway this registration would not live here

Reply:
Yeah I can remove the methods that rely on it. We'll have to use it somewhere though, but I can move that to baseapp where we already require comet.